src/index.ts

```diff
@@ -149,46 +189,66 @@ app.get('/repodetails/:loginName', authenticateToken, async (req: Request, res:
 });
 
 
-app.get('/runnertoken/:org/:reponame', authenticateToken,  async (req:Request, res:Response) => {
-    const ORG = req.params.org;
-    const REPONAME = req.params.reponame;
+app.get('/runnertoken/:org/:reponame/:accesstoken',  async (req:Request, res:Response) => {
+    //const ORG = req.params.org;
+    //const REPONAME = req.params.reponame;
 
-    // initalize octokit
-    const instance = appManager.getOctokitInstance(ORG);
-    var octo:any = null;
-    var installationID: number = 0;
-    if (instance.length != 0){
-        installationID = instance[0].id;
-        octo = instance[0].octokit;
-    }
+    // get the token
+    const token = req.params.accesstoken;
 
-    // get access token for octo
-    //const { token } = await octo.data ({ installationId: installationID, permissions: { admin: "org" } });
-    const { data: { token } } = await octo.request("POST /app/installations/{installation_id}/access_tokens", {
-        installation_id: installationID,
-        permissions: {
-          administration: "write",
-        },
-      });
-    
-    // get token for runner based on org and reponame
-    await octo.request('POST /repos/{owner}/{repo}/actions/runners/registration-token', {
-        owner: ORG,
-        repo: REPONAME,
-        headers: {
-            'X-GitHub-Api-Version': '2022-11-28',
-            authorization: `token  ${token}`,
-            accept: "application/vnd.github.machine-man-preview+json",
+    // decode the token, extract org and reponame
+    const secretKey = await fs.readFile(path.join(__dirname, 'src/middleware/custom-token.pem')) 
+    const decoded:JwtPayload =  <JwtPayload> jwt.verify(token, secretKey);
+    const ORG = decoded.org
+    const REPONAME = decoded.reponame;
+
+    // check if access token is correct
+    const tokenMap = appManager.getAccessTokens()
+    if (tokenMap[ORG+" "+REPONAME] == req.params.accesstoken){
```

Comment:
Still confused: What is the rationale for `tokenMap`? Doesn't this ensure that at most one runner can be queued for each repo at a time?

Reply:
Understood. I can use two approaches - 
1) Keep the tokenMap, but add a time stamp component to it and pass that timestamp to compute service as well. When the request comes back to the github service, we can use org, repo and timestamp (all decoded from the token) and compare it in the map

2) compare the decoded org and repo from token, to the or and repo passed in URL